backend/inspirehep/orcid/utils.py

```diff
@@ -226,3 +232,38 @@ def distributed_lock(lock_name, expire=10, auto_renewal=True, blocking=False):
 
 class DistributedLockError(Exception):
     pass
+
+
+def update_moved_orcid(old_orcid, new_orcid):
+    author_record = AuthorsRecord.get_record_by_pid_value(old_orcid, "orcid")
+    if new_orcid not in get_value(author_record, "ids.value", []):
+        new_author_ids = [
+            {"schema": "ORCID", "value": new_orcid},
+            *author_record["ids"],
+        ]
+        author_record["ids"] = new_author_ids
+        author_record.update(dict(author_record))
+        db.session.commit()
```

Comment:
why are you committing here? I think it's best to run this whole thing in a single transaction as we don't want to partially apply these changes.